### ISSUE TITLE
Video sprites condition switch stop working when timer

### ISSUE THREAD

**Issue by GGAdams3D** [renpy/renpy#4492]
Hi again, still the same that wants to make video sprites. It's crazy how complicated it is. So this time I have a new issue: I want the sprites to switch from talking to idle after a specific time so that the animation changed smoothly rather than suddenly when the texts is written.

Here is a video preview of the issue: 
https://user-images.githubusercontent.com/128613924/228638783-c6877080-7bfc-41da-9068-7cc91ddd263b.mp4

My idea for this issue is with timers and callbacks. The sprites use a condition switch from a variable that is true when the character start speaking, then it also set to 2 secondes the timer (which the time of the animation sprite to loop fully) and another timer that will change the variable if the timer is 0 and and character speaking is the character.

**Here is some code:**

_**The sprite condition switch:**_
`image natsu neutral = ConditionSwitch(
    "character_is_talking != 'natsu'", "natsu_neutral_[natsu_costume]", "character_is_talking == 'natsu'", "natsu_neutral_[natsu_costume]_talking")`

_**The timers:**_
`init:
    $ timer_range = 0
    $ natsu_sprite = 0
    $ time_natsu = 0

screen countdown:
    timer 0.1 repeat True action If(time_natsu > 0, true=SetVariable('time_natsu', time_natsu - 0.1))
    timer 0.1 repeat True action If(character_is_talking == 'natsu' and time_natsu <= 0, true=SetVariable('character_is_talking', 'none'))`

**_And the callback:_**
`def name_callback(event, interact=True, name=None, **kwargs):
        global speaking_char
        if event == "begin":
            speaking_char = name
        if event == "show":
            renpy.music.play("audio/beeps.mp3", channel="sound", loop="True")
        elif event == "slow_done" or event == "end":
            renpy.music.stop(channel="sound")

        global character_is_talking
        if event == "begin":
            character_is_talking = name

        global time_natsu
        if event == "begin":
            time_natsu = 2`

So I really don't know why the timer make it stop the video. I noticed that it depends of the speed of the timer, if too fast it will looks like stuck, and it's only the timer that has the condition involved (so the second one).

**Comment by GGAdams3D** [comment]
bump

**Comment by renpytom** [comment]
I'm not able to repeat this problem in the current Ren'Py, but I'm not completely sure I'm replicating it right. 

If you can still reproduce this, please make a new game with just the relevant files and code - I don't want the entire project, just the smallest possible replication that causes the problem, so I don't have to spend more time trying to get the problem to even occur, or go down blind alleys trying to figure out what's going on.

**Comment by GGAdams3D** [comment]
> I'm not able to repeat this problem in the current Ren'Py, but I'm not completely sure I'm replicating it right.
> 
> If you can still reproduce this, please make a new game with just the relevant files and code - I don't want the entire project, just the smallest possible replication that causes the problem, so I don't have to spend more time trying to get the problem to even occur, or go down blind alleys trying to figure out what's going on.

Sure, I can send you the demo with the bug on. Also I use a nightbuild as you made the update for the video sprite channels. The code is pretty simple and in the main script.

**Comment by renpytom** [comment]
Not the bug demo from last time - please break it out into something small that shows the problem without tons of extraneous script.

**Comment by GGAdams3D** [comment]
> 

My demo is already very small. I don't get what you mean. I can remove some scripts to make it more easy to navigate.

**Comment by renpytom** [comment]
Yes, please. There is a lot of script here that is not obviously relevant.

**Comment by GGAdams3D** [comment]
> Yes, please. There is a lot of script here that is not obviously relevant.

I sended you the new demo. I only kept what matters and added explanations. I hope it would be clear enough. Looking forward your help.

**Comment by renpytom** [comment]
Basically, using a timer like you did is something of a bad idea. I think the problem is the talking stopped .1 seconds late, and that probably was because the first timer would deduct .1 seconds, but the data won't actually update, until the second went off.

```renpy
screen countdown:
    #The first timer is for the integer variable that defines the time
    timer 0.1 repeat True action If(time_natsu > 0, true=SetVariable('time_natsu', time_natsu - 0.1))
    #The second timer check the condition until the integer is 0 and also this the correct character speaking
    timer 0.1 repeat True action If(character_is_talking == 'natsu' and time_natsu <= 0, true=SetVariable('character_is_talking', 'none'))
```

I'd suggest just use a counter that takes the duration to wait for:

```renpy
screen countdown(dur):
    timer dur action SetVariable('character_is_talking', 'none')
```

And then use renpy.show_screen to show it. 

```renpy
       if event == "begin":
            renpy.show_screen("countdown", dur=2, _transient=True)
```

**Comment by GGAdams3D** [comment]
Thanks for the help. Indeed I didn't think of my idea was good but I didn't have better. I didn't really understand why it shouldn't work. So thanks for the counter idea, I'll try it soon enough and if I'll go edit my post if it worked! Thanks again.

**Comment by GGAdams3D** [comment]
> 

Okay so I tried it and make it work. It works as intended, but it's really resolved. The sprites still can have a little jump from one to another. It's not smooth, it's like it the sprite start is not perfectly sync to the timer it seems.

Is there a way to make the sprites queue? Or sync exactly the timer to a video? Or defined the time of the video length as the duration variable (could be too complexe tho)?
